python_modules/dagster/dagster/core/execution_plan/simple_engine.py

```diff
@@ -73,6 +75,20 @@ def start_inprocess_executor(
 
             step_context = pipeline_context.for_step(step)
 
+            failed_inputs = [
+                step_input.prev_output_handle.step_key
+                for step_input in step.step_inputs
+                if step_input.prev_output_handle.step_key in propogated_step_failures
+            ]
+            if failed_inputs:
+                step_context.log.info(
+                    ('Dependencies for step {step} failed: {failed_inputs}. Not executing.').format(
+                        step=step.key, failed_inputs=failed_inputs
+                    )
+                )
+                propogated_step_failures.add(step.key)
+                continue
```

Comment:
I think we might want to make this `failed_or_skipped_steps` and add to the list in the uncovered_inputs case below as well